src/v/serde/parquet/encoding.h

```diff
@@ -30,4 +36,15 @@ iobuf encode_plain(chunked_vector<byte_array_value> vals);
 
 iobuf encode_plain(chunked_vector<fixed_byte_array_value> vals);
 
+// Levels (definition and repetition) are encoded using Parquet's hybrid
+// run-length encoding/bitpacking schema. Bit packing requires the maximum
+// value to be known in advance.
+//
+// See:
+// https://parquet.apache.org/docs/file-format/nestedencoding/
+// https://parquet.apache.org/docs/file-format/data-pages/encodings/#run-length-encoding--bit-packing-hybrid-rle--3
+//
+// If `levels` is empty `max_value` should be `0`.
+iobuf encode_levels(uint8_t max_value, const chunked_vector<uint8_t>& levels);
```

Comment:
Just wondering if this gets called infrequently enough to just compute max_value from levels?

Reply:
I was planning on computing this in a single pass as we receive records. Most parquet implementations buffer records computing stats incrementally, then flush the row group. This is called in the flush method, so we should have max value already computed